tests/trogdor/trogdor_utils.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import array
+import re
+import sys
+import json
+
+from confluent_kafka.cimpl import KafkaException, KafkaError, Producer
+
+from confluent_kafka.admin import AdminClient, NewTopic
+
+
+def partition_set(topics):
+    """
+    Return an array of TopicPartition set, so we can send messages to them one by one.
+    For example, for topics
+    {
+      "foo" : { "numPartitions": 3 }
+      "bar" : { "numPartitions": 2 }
+    }
+    return
+    [("foo", 0), ("foo", 1), ("foo", 2), ("bar", 0), ("bar", 1)]
+    """
+    topic_partitions = []
+    for t_name in topics:
+        t = topics[t_name]
+        nr_partitions = t.get("numPartitions", 1)
+        for i in range(0, nr_partitions):
+            topic_partitions.append((t_name, i))
+    return topic_partitions
+
+
+def expand_topics(topics):
+    """
+    Return extended topics.
+    For example, for topics { "foo[1-3]" : {} }, return { "foo1" : {}, "foo2" : {}, "foo3" : {}}
+    """
+    topic_expand_matter = re.compile(r'(.*?)\[([0-9]*)\-([0-9]*)\](.*)')
```

Comment:
the lo and hi integers should be required, and no need to escape -, thus:
`re.compile(r'(.*?)\[([0-9]+)\-([0-9]+)\](.*)')`

Reply:
Good catch, will update the Java version too.